resolver/src/main/java/io/netty/resolver/HostsFileParser.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver;
+
+import io.netty.util.NetUtil;
+import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.netty.util.internal.ObjectUtil.*;
+
+/**
+ * A parser for hosts files.
+ */
+public final class HostsFileParser {
+
+    private static final String WINDOWS_HOSTS_FILE_PATH = "C:\\Windows\\system32\\drivers\\etc\\hosts";
+    private static final String X_PLATFORMS_HOSTS_FILE_PATH = "/etc/hosts";
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(HostsFileParser.class);
+
+    /**
+     * Parse hosts file at standard OS location.
+     *
+     * @return a map of hostname or alias to {@link InetAddress}
+     */
+    public static Map<String, InetAddress> parseSilently() {
+        String path = PlatformDependent.isWindows() ? WINDOWS_HOSTS_FILE_PATH : X_PLATFORMS_HOSTS_FILE_PATH;
+        try {
+            return parse(new File(path));
+        } catch (IOException e) {
+            logger.error("Failed to load and parse hosts file", e);
```

Comment:
Also, please add the actual path to the hosts file to the log message. e.g.

    "Failed to load and parse hosts file: {}", path, e